doc/release/migration-guide-18.6.md

```diff
@@ -3,7 +3,13 @@
 > [!NOTE]
 > This is a working draft for the up-coming v18.6.0 release.
 
-This document lists recommended and required changes for those migrating from the previous v18.5.0 firmware release to the new v18.6.0 firmware release.
+This document lists recommended and required changes for those migrating from the previous v18.5.0
+firmware release to the new v18.6.0 firmware release.
+
+* The upstream (draft) migration guide for the Zephyr v4.2.0 release is available
+[here](https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/migration-guide-4.2.rst).
+  * Notable topics include a switch from Python 3.10 to Python 3.12 as the minimum version for
```

Comment:
We are actually rolling this back to 3.10 (because some distributions don't package 3.12): https://github.com/zephyrproject-rtos/zephyr/pull/92555

Reply:
Upstream? Really?